### ISSUE TITLE
Bug - Table - accessibility rules have changed, all thead cells should be `<th>`s

### ISSUE THREAD

**Issue by mattnolting** [patternfly/patternfly#6272]
**Describe the problem**
Accessibility rules have changed, `<td>`s in `thead`s are no longer acceptable. 

**How do you reproduce the problem?**
Test with lighthouse, other accessibility tools, and/or screen reader

**Expected behavior**
Columns are described clearly and concisely with text and without altering table layout

**Is this issue blocking you?**
- [x] Replace all `thead > td`s (blank visually) to `<th><span class="pf-v5-screen-reader">{descriptive text}{/span></th>`
- [x] For non-text content (select all checkbox, expand all buttons), `<th aria-label="Row check">{interactive element such as input or button with its own aria-label}</th>`
- [x] Update `:empty` table cell styling to be something like 
```
:is(.pf-v5-c-table__th, .pf-v5-c-table__td) {
  &:has(.pf-v5-screen-reader),
  &:empty {
    width: auto
    min-width: auto;
    padding: 0;
  }
}
```

**Comment by mcoker** [comment]
closed by https://github.com/patternfly/patternfly/pull/6689
